src/plugins/intel_gpu/include/intel_gpu/runtime/utils.hpp

```diff
@@ -183,4 +189,97 @@ inline std::string to_string(const T& v) {
 /// @}
 /// @endcond
 /// @}
+
+/// @brief LRU cache which remove the least recently used data when cache is full.
+template<typename TypeK, typename TypeD>
+class LRUCache {
+public:
+    struct CacheEntry {
```

Comment:
BTW, I suggest checking if we can make `kernels_cache` a part of `LRUCache` and use it in the program even for static scenario (especially if we combine it with unlimited capacity mode for static scenario)

Reply:
Okay, I see. I will check how to modify kernels_cache using LRUCache

Reply:
That's good idea. As your comments below, I will make new PR for this idea after this PR is merged. thanks.